polygon-zkevm-adaptor/tests/end_to_end.rs

```diff
@@ -276,6 +335,180 @@ async fn test_preconfirmations() {
+    tracing::info!("waiting for batches to be verified");
+    sleep(Duration::from_secs(5)).await;
+
+    // Force an L1 reorg.
+    tracing::info!("causing L1 reorg");
+    reorgme.join();
+
+    // Wait a bit for the nodes to recover.
+    sleep(Duration::from_secs(10)).await;
+
+    // Send another transaction to ensure the nodes are still syncing.
+    let txn_hash = l2
+        .send_transaction(
+            TransactionRequest {
+                from: Some(l2.inner().address()),
+                to: Some(Address::zero().into()),
+                value: Some(transfer_amount),
+                ..Default::default()
+            },
+            None,
+        )
+        .await
+        .unwrap()
+        .tx_hash();
+    let submitted = Instant::now();
+    tracing::info!("Sent transaction {txn_hash:?} at {submitted:?}");
+
+    // Wait for the transaction to complete on L2, using both the regular RPC and the
+    // preconfirmation RPC in parallel.
+    futures::join!(
+        await_transaction(&l2_preconf, txn_hash),
+        await_transaction(&l2, txn_hash),
+    );
+
+    // Check the effects of the transfer.
+    assert_eq!(
+        l2.get_balance(l2.inner().address(), None).await.unwrap(),
+        l2_initial_balance - transfer_amount * 2
+    );
+    assert_eq!(
+        l2_preconf
+            .get_balance(l2_preconf.inner().address(), None)
+            .await
+            .unwrap(),
+        l2_initial_balance - transfer_amount * 2
+    );
+
+    // Wait for the verified batches to catch up, to be sure everything is still syncing properly.
+    // This forces the test to run long enough, and for the nodes to sync enough verified batches,
+    // that it will be clearly visible in the logs if the preconfirmations node is failing to sync
+    // verified batches, which is the problem we saw when there was a reorg in production.
+    //
+    // Note that the test won't necessarily fail if the preconfirmations node is unable to sync
+    // verified batches, because syncing these batches doesn't actually affect the observable state
+    // of the preconfirmations node: it's state is only affected by blocks that it syncs directly
+    // from HotShot, asynchronously with respect to the verified state. But at least we will be able
+    // to check the logs for issues if we are actively working on reorg handling.
+    let l2_height = sequencer
+        .get::<u64>("status/latest_block_height")
+        .send()
+        .await
+        .unwrap();
+    let verified_filter = rollup.verify_batches_trusted_aggregator_filter();
+    loop {
+        tracing::info!("waiting for batch {l2_height} to be verified");
+        let event = verified_filter
+            .stream()
+            .await
+            .unwrap()
+            .next()
+            .await
+            .unwrap()
+            .unwrap();
+        tracing::info!("current verified batch is {}", event.num_batch);
+        if event.num_batch >= l2_height {
```

Comment:
Should this time out?

Reply:
Yeah but I'd rather handle it at the CI job config level (currently has a 1 hour timeout for the entire slow tests workflow)